src/dirs.rs

```diff
@@ -1,21 +1,71 @@
+#[cfg(not(target_os="android"))]
 use appdirs;
 
+use std::fs;
 use std::path::PathBuf;
 
+#[cfg(target_os="android")]
+static mut ANDROID_DATA_DIR: Option<PathBuf> = None;
+
 pub fn get_config_dir() -> Result<PathBuf, ()> {
-    let mut dir = appdirs::user_config_dir(Some("activitywatch"), None, false)?;
-    dir.push("aw-server-rust");
-    return Ok(dir);
+    #[cfg(not(target_os="android"))]
+    {
+        let mut dir;
+        dir = appdirs::user_config_dir(Some("activitywatch"), None, false)?;
+        dir.push("aw-server-rust");
+        return Ok(dir);
+    }
+
+    #[cfg(target_os="android")]
+    {
+        return Err(());
+    }
 }
 
 pub fn get_data_dir() -> Result<PathBuf, ()> {
-    let mut dir = appdirs::user_data_dir(Some("activitywatch"), None, false)?;
-    dir.push("aw-server-rust");
-    return Ok(dir);
+    #[cfg(not(target_os="android"))]
+    {
+        let mut dir = appdirs::user_data_dir(Some("activitywatch"), None, false)?;
+        dir.push("aw-server-rust");
+        return Ok(dir);
+    }
+
+    #[cfg(target_os="android")]
+    unsafe {
+         return match ANDROID_DATA_DIR {
+            Some(ref path) => Ok(path.to_path_buf()),
+            None => Err(())
+        }
```

Comment:
@johan-bjareholt Any better way to do this? The current way is having a static variable that should be set before calling it on Android. 

Maybe get the android_data_dir from Rust on every call? Not sure how though. (But could add a comment about it)

Reply:
For now you could just hardcode it to "/data/data/org.activitywatch.activitywatch/".

The right thing to do I believe though is to figure out the android packagename and set it to "/data/data/packagename", but not sure how to actually do that from rust in a nice way but there seems to be possible in multiple ways, here's one.
https://stackoverflow.com/questions/42918762/how-to-get-app-package-name-or-applicationid-using-jni-android
But at that point, maybe it's easier to just pass it down from Java.

Also, appdirs-rs says "Android support should be coming soon" but that repo hasn't been updated since 2016.

Reply:
I'd rather not hardcore it, and can't be bothered to try and call the JNI from Rust right now, so this'll have to do.

That being said, I noticed they were calling the Android logger using the JNI, which I'd really like to be able to. Should look into that.